### PR TITLE
Make `run` and `cron` integration more stable

### PR DESCRIPTION
**Pull request by apfelbox**
| Q             | A
| ------------- | --------------------------------------------------------------------- |
| Bug fix?      | yes                                                               |
| New feature?  | no                    |
| Improvement?  | no     |
| BC breaks?    | no                                                                |
| Deprecations? | no     |
| Docs PR       | —                                   |

### CODE REVIEW

File: bin/cron [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
+$php = "/usr/bin/env php";
+
+// try possible values for a specialized PHP version
+$possibleExecutables = [
+    '/usr/local/bin/php',
+];
+
+foreach ($possibleExecutables as $possibleExecutable)
+{
+    $result = trim(shell_exec("command -v {$possibleExecutable}"));
```
Comment (keichinger) [inline bin/cron]:
```suggestion
    $result = \trim(\shell_exec("command -v {$possibleExecutable}"));
```

File: bin/cron [hunk 2]
```diff
@@ -1,3 +1,23 @@
...
+
+    if ("" !== $result)
+    {
+        $php = $result;
+        break;
+    }
+}
+
+passthru("{$php} " . \escapeshellarg(__DIR__ . "/run.php"));
```
Comment (keichinger) [inline bin/cron]:
```suggestion
\passthru("{$php} " . \escapeshellarg(__DIR__ . "/run.php"));
```

File: bin/run.php [hunk 3]
```diff
@@ -15,7 +15,9 @@
     exit(2);
 }
 
-passthru("{$projectDir}/bin/console cron:run --ansi", $returnVar);
+$php = escapeshellcmd(PHP_BINARY);
```
Comment (keichinger) [inline bin/run.php]:
```suggestion
$php = \escapeshellcmd(\PHP_BINARY);
```

File: bin/run.php [hunk 4]
```diff
@@ -15,7 +15,9 @@
...
+$console = escapeshellarg("{$projectDir}/bin/console");
```
Comment (keichinger) [inline bin/run.php]:
```suggestion
$console = \escapeshellarg("{$projectDir}/bin/console");
```

File: bin/run.php [hunk 5]
```diff
@@ -15,7 +15,9 @@
...
+passthru("{$php} {$console} cron:run --ansi", $returnVar);
```
Comment (keichinger) [inline bin/run.php]:
```suggestion
\passthru("{$php} {$console} cron:run --ansi", $returnVar);
```

**Comment by apfelbox** [comment]
All these CS changes don't give you anything, as the file has no namespace...

**Pull request merged**